crates/resolver-tests/src/lib.rs

```diff
@@ -227,51 +242,201 @@ fn sat_at_most_one(solver: &mut impl varisat::ExtendFormula, vars: &[varisat::Va
 }
 
 fn sat_at_most_one_by_key<K: std::hash::Hash + Eq>(
-    cnf: &mut impl varisat::ExtendFormula,
+    solver: &mut varisat::Solver<'_>,
     data: impl Iterator<Item = (K, varisat::Var)>,
 ) -> HashMap<K, Vec<varisat::Var>> {
-    // no two packages with the same links set
+    // no two packages with the same keys set
     let mut by_keys: HashMap<K, Vec<varisat::Var>> = HashMap::new();
     for (p, v) in data {
         by_keys.entry(p).or_default().push(v)
     }
     for key in by_keys.values() {
-        sat_at_most_one(cnf, key);
+        sat_at_most_one(solver, key);
     }
     by_keys
 }
 
+fn find_compatible_dep_summaries_by_name_in_toml(
+    pkg_dependencies: &[Dependency],
+    by_name: &HashMap<InternedString, Vec<Summary>>,
+) -> IndexMap<InternedString, Vec<Summary>> {
```

Comment:
Why does this need to be in `IndexMap`? Not objecting to it, it's a good library, I'm just not sure why we need to keep the order.

Reply:
I have currently this code which depends of the iteration order of the map:
```rust
for (dep, compatible_dep_summaries) in pkg_dependencies
        .iter()
        .zip(compatible_dep_summaries_by_name_in_toml.values())
```
But this is not necessary since I can directly use the key, so I will use an `HashMap` here.